scylla/src/network/connection.rs

```diff
@@ -900,6 +902,30 @@ impl Connection {
         .await
     }
 
+    fn handle_result_metadata_new_id(
+        prepared_statement: &PreparedStatement,
+        query_response: &QueryResponse,
+    ) {
+        if let ResponseWithDeserializedMetadata::Result(ResultWithDeserializedMetadata::Rows(
+            rows_result,
+        )) = &query_response.response
+        {
+            if rows_result.0.metadata().id().is_some()
+                && rows_result.0.metadata().id()
+                    != prepared_statement.get_current_result_metadata().id()
```

Comment:
:wrench: Can we use `match` with guard for clarity here?

Reply:
Can `id()` be `None` on `PreparedStatement`? When can this happen? Is it possible that we're here and it's `None`?